### DISCUSSION TITLE
Mac Platform API Support

### DISCUSSION THREAD

**Discussion by leosimanonok** [dotnet/maui#10334] [Q&A]
The app I'm working on needs to be able to save files to a specific folder on a user's device. We've been using Cocoa through VS to access the NSSavePanel class and the security bookmark functionality. Does support for file system access on Mac devices exist in MAUI/ will it be supported down the line? If not, are there any workarounds that you suggest?

**Comment by xtuzy** [comment]
i google find this https://stackoverflow.com/questions/72751200/nssavepanel-for-mac-catalyst

**Comment by noobsoftware** [comment]
It seems its possible to include AppKit using this library: https://github.com/mhdhejazi/Dynamic It would be necessary to create a binding for this. In Visual Studio 2022 you have the option for creating a binding from objective-c but this library is built in Swift, not sure exactly how that works.

**Comment by noobsoftware** [comment]
When attempting to build a project referencing a Mac Catalyst binding project built around objective-c code i get the following error:

maui /usr/local/share/dotnet/packs/Microsoft.MacCatalyst.Sdk/15.4.2372/targets/Xamarin.Shared.Sdk.targets(3,3): Error: clang++ exited with code 1: ld: building for Mac Catalyst, but linking in dylib built for macOS, file dylib for architecture x86_64 clang: error: linker command failed with exit code 1 (use -v to see invocation) (test)

I've tried using a .framework built for multiplatform and for ios i get a similar error. This seems to be bug.

**Reply by JoeTorrey** [reply]
I don’t think I have any more money left to buy a new one but I will check with my dad to see if he has any money left for me to buy one of these for me and if he does I will get it from you and you can just give me the money back when you get home
